augur/io/metadata.py

```diff
@@ -158,12 +167,9 @@ def read_table_to_dict(table, duplicate_reporting=DataErrorMethod.ERROR_FIRST, i
         try:
             # Note: this sort of duplicates _get_delimiter(), but it's easier if
             # this is separate since it handles non-seekable buffers.
-            dialect = csv.Sniffer().sniff(table_sample, VALID_DELIMITERS)
-        except csv.Error as err:
-            raise AugurError(
-                f"Could not determine the delimiter of {table!r}. "
-                "File must be a CSV or TSV."
-            ) from err
+            dialect = csv.Sniffer().sniff(table_sample, valid_delimiters)
+        except csv.Error:
+            raise InvalidDelimiter
```

Comment:
Is `csv.Error` here used for other error conditions too?

Reply:
The `csv.Sniffer.sniff()` [source code](https://github.com/python/cpython/blob/b57105ae33e1f61e6bdf0eec45c4135d067b9b22/Lib/csv.py#L180-L206) shows that (as of now) this is only raised when the delimiter cannot be determined:

```py
if not delimiter:
    raise Error("Could not determine delimiter")
```

I can add a check that the `csv.Error` matches `Could not determine delimiter` before raising `InvalidDelimiter`, otherwise raise the original error. Does that sound better to you?

Or, maybe "Could not determine delimiter" does not imply "invalid delimiter" for reasons that I can't think of right now. In that case, it might be better to rename the error class to `IndeterminateDelimiter` or something similar. The existing error handling added in this PR already use the wording "Could not determine the delimiter" which is in line with the original error reason.

Reply:
Hmm. I think what you have now is fine, if you want to keep it, given the current use of `Error`. I don't know how to judge if it's more likely that `Sniffer` starts raising additional `Error`s (and we accidentally mask them as delimiter issues) or if the exact `Error` description changes (and we accidentally stop catching delimiter issues).

Reply:
Keeping as-is and adding a comment in 2bbf9288fdb38e969bfd838a3f9aaa6287973bd9.